### PR TITLE
en-croissant: init at 0.9.2

### PR DESCRIPTION
**Pull request by lcscosta**
## Description of changes

Added the [en-croissant](https://www.encroissant.org/) a Modern Chess Database .
This is an app built with [Tauri](https://github.com/tauri-apps/tauri).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/games/en-croissant/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    sha256 = "1lrl0pg0aihx87yqygfailbf9n9ra1l0a7zg40j36c92h8k0zmcb";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    dpkg
+  ];
+
+  buildInputs = [
+    openssl_1_1
```
Comment (kirillrdy) [inline pkgs/games/en-croissant/default.nix:27]: openssl_1_1 is deprecated and in the process of being removed from nixpkgs https://github.com/NixOS/nixpkgs/issues/269713

File: pkgs/games/en-croissant/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+, openssl_1_1
+}:
+
+stdenv.mkDerivation rec {
+  name = "en-croissant";
+  version = "0.7.1";
+
+  src = fetchurl {
+    url = "https://github.com/franciscoBSalgueiro/en-croissant/releases/download/v${version}/en-croissant_${version}_amd64.deb";
+    sha256 = "1lrl0pg0aihx87yqygfailbf9n9ra1l0a7zg40j36c92h8k0zmcb";
```
Comment (kirillrdy) [inline pkgs/games/en-croissant/default.nix:18]:
```suggestion
    hash = "1lrl0pg0aihx87yqygfailbf9n9ra1l0a7zg40j36c92h8k0zmcb";
```

File: pkgs/games/en-croissant/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+, webkitgtk
+, libsoup
+, openssl_1_1
+}:
+
+stdenv.mkDerivation rec {
+  name = "en-croissant";
+  version = "0.7.1";
+
+  src = fetchurl {
```
Comment (kirillrdy) [inline pkgs/games/en-croissant/default.nix:16]: building from source is preferred approach in nixpkgs

https://github.com/franciscoBSalgueiro/en-croissant#building-from-source

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -37738,6 +37738,8 @@ with pkgs;
 
   EmptyEpsilon = callPackage ../games/empty-epsilon { };
 
+  en-croissant = callPackage ../games/en-croissant { };
```
Comment (kirillrdy) [inline pkgs/top-level/all-packages.nix:37741]: new packages should be added into pkgs/by-name 

you can read more about it https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md

File: pkgs/games/en-croissant/default.nix (hunk 3 above)
Comment (kirillrdy) [reply pkgs/games/en-croissant/default.nix:16]: feel free to reach out if you run into issues

**Comment by lcscosta** [comment]
The CI is being extremely cruel to me, perhaps the reviewers can help me in this battle.

File: pkgs/by-name/en/en-croissant/package.nix [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+      moreutils
+      nodePackages.pnpm
+      cacert
+    ];
+
+    installPhase = ''
+      export HOME=$(mktemp -d)
+      pnpm config set store-dir $out
+      # use --frozen-lockfile to avoid checking git deps
+      pnpm install --frozen-lockfile
```
Comment (kirillrdy) [inline pkgs/by-name/en/en-croissant/package.nix:49]: I cant see where the lock file comes from ?

File: pkgs/by-name/en/en-croissant/package.nix [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+      rm -rf $out/v3/tmp
+      for f in $(find $out -name "*.json"); do
+        sed -i -E -e 's/"checkedAt":[0-9]+,//g' $f
+        jq --sort-keys . $f | sponge $f
+      done
+    '';
+
+    dontFixup = true;
+    outputHashMode = "recursive";
+    outputHash = "sha256-DTsT30o2BcXqqQW7uKtIexGoPT1KTxqPgDxOh+16olY=";
```
Comment (kirillrdy) [inline pkgs/by-name/en/en-croissant/package.nix:61]: you should use something like `fetchNpmDeps` then you wont need to deal with fixed output derivations yourself

Comment (lcscosta) [reply pkgs/by-name/en/en-croissant/package.nix:61]: Hi @kirillrdy, we need to create a `fetchPnpmDeps`. Unfortunately, this project uses only pnpm as its node-modules manager. The nix-packages with pnpm as node manager will be a little bit hard to read/maintain with this necessary workaround for downloading node-modules.

Another examples with the same implementation logic:

pkgs/applications/misc/pot/default.nix
pkgs/applications/networking/geph/default.nix
pkgs/by-name/ve/vesktop/package.nix

File: pkgs/by-name/en/en-croissant/package.nix (hunk 5 above)
Comment (lcscosta) [reply pkgs/by-name/en/en-croissant/package.nix:49]: This lock file refer to the `pnpm-lock.yaml` that is in the fetched Github repository.

Comment (kirillrdy) [reply pkgs/by-name/en/en-croissant/package.nix:49]: i see, `src` gets extracted, I missed that bit

File: pkgs/by-name/en/en-croissant/package.nix (hunk 6 above)
Comment (kirillrdy) [reply pkgs/by-name/en/en-croissant/package.nix:61]: ok i see

as you can see from ofborg, the hash is different. The difficulty with fixed output derivations is that they must always produce same output. I get same hash as ofborg ( this is on x86_64-linux ) we'll need to figure out why content is different and try to fix it.

Comment (lcscosta) [reply pkgs/by-name/en/en-croissant/package.nix:61]: > ok i see
> 
> as you can see from ofborg, the hash is different. The difficulty with fixed output derivations is that they must always produce same output. I get same hash as ofborg ( this is on x86_64-linux ) we'll need to figure out why content is different and try to fix it.

Hmmmm, I believe that this problem are from the optional dependencies, like rollup and esbuild. I will work on a way to use the `nodePackages.<package-name>` in the build, and in this way we can ignore the download of optional dependencies with pnpm.

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -37738,6 +37738,8 @@ with pkgs;
 
   EmptyEpsilon = callPackage ../games/empty-epsilon { };
 
+  en-croissant = callPackage ../by-name/en/en-croissant/package.nix { };
+
```
Comment (Misterio77) [inline pkgs/top-level/all-packages.nix:37742]: AFAIK, you won't need this when using `by-name`

**Review by Misterio77 — COMMENTED** [review]
Builds as expected and the package seems to work nicely on my end.

image

No hash mismatch at all (tried both x86-64-linux and aarch64-linux):
```
$ nix build github:nixos/nixpkgs/pull/271493/head#en-croissant.pnpm-deps --print-out-paths 
/nix/store/59nsarxf4sly419as0b9sgd0fn2cvv8q-en-croissant-pnpm-deps-0.8.0

$ nix build github:nixos/nixpkgs/pull/271493/head#legacyPackages.aarch64-linux.en-croissant.pnpm-deps --print-out-paths
/nix/store/59nsarxf4sly419as0b9sgd0fn2cvv8q-en-croissant-pnpm-deps-0.8.0

**Comment by lcscosta** [comment]
@ofborg build en-croissant

File: pkgs/top-level/all-packages.nix (hunk 7 above)
Comment (kirillrdy) [reply pkgs/top-level/all-packages.nix:37742]: correct, this is not needed

**Comment by kirillrdy** [comment]
@lcscosta to get CI passing "pkgs/by-name" check you need to rebase on latest master

**Comment by lcscosta** [comment]
> @lcscosta to get CI passing "pkgs/by-name" check you need to rebase on latest master

@kirillrdy, I fixed the 'pkgs/by-name' task and am currently awaiting the evaluation by ofborg. @Misterio77 helped me in testing the hash mismatch (aarch64 not worked on his machine), I am not confident that I have completely resolved this problem. If the pnpm patch does not work, I plan to try removing the optional dependencies, as I mentioned in the comment above.

**Comment by kirillrdy** [comment]
@lcscosta can you please either squash commits or update commit messages to match contribution guidelines ?

**Comment by kirillrdy** [comment]
@Misterio77 if you have access to both fixed output derivations you can just `diff -r` them to see whats different

**Comment by Misterio77** [comment]
Hey @lcscosta and @kirillrdy.

Here's the x86_64-linux -> aarch64-linux diff:

```diff
diff '--color=auto' -r /nix/store/ggnzrp25am692b887qcidb5bb7qbc6x2-en-croissant-pnpm-deps-0.8.0/v3/files/05/eaedbfcc4e89510bcf9ada0761655c33cab7e9bde0c4372600dfb156eba8da4b53df94136135f062a32cac3e7775db7983ea835a73c90286b2bcdbc3ea4f6b-index.json /nix/store/s8xvcm8n4dk2qi6p4hrpcr07k0sx1wkr-en-croissant-pnpm-deps-0.8.0/v3/files/05/eaedbfcc4e89510bcf9ada0761655c33cab7e9bde0c4372600dfb156eba8da4b53df94136135f062a32cac3e7775db7983ea835a73c90286b2bcdbc3ea4f6b-index.json
60c60
<     "linux-x64-node-v18-{\"/@fortawesome/fontawesome-common-types/6.4.0\":{}}": {
---
>     "linux-arm64-node-v18-{\"/@fortawesome/fontawesome-common-types/6.4.0\":{}}": {
diff '--color=auto' -r /nix/store/ggnzrp25am692b887qcidb5bb7qbc6x2-en-croissant-pnpm-deps-0.8.0/v3/files/1c/d8a2d77db17e89a0e50559c341f05d7a4ddb6b3b5411d006ff4278ebbdb5f26786c59035dda5993362f024ab9f0e41fb960e9b3bb55283c0cb1697b09e9181-index.json /nix/store/s8xvcm8n4dk2qi6p4hrpcr07k0sx1wkr-en-croissant-pnpm-deps-0.8.0/v3/files/1c/d8a2d77db17e89a0e50559c341f05d7a4ddb6b3b5411d006ff4278ebbdb5f26786c59035dda5993362f024ab9f0e41fb960e9b3bb55283c0cb1697b09e9181-index.json
30c30
<     "linux-x64-node-v18-{}": {
---
>     "linux-arm64-node-v18-{}": {
diff '--color=auto' -r /nix/store/ggnzrp25am692b887qcidb5bb7qbc6x2-en-croissant-pnpm-deps-0.8.0/v3/files/4c/a151a7d4f1ac90dd4567d2b12111284a2f9babf7d221e7ae395c1a12c13ad45e136f18a755520517cedc64da385328c6c35eaf1e138eb8734ae99af616bcdd-index.json /nix/store/s8xvcm8n4dk2qi6p4hrpcr07k0sx1wkr-en-croissant-pnpm-deps-0.8.0/v3/files/4c/a151a7d4f1ac90dd4567d2b12111284a2f9babf7d221e7ae395c1a12c13ad45e136f18a755520517cedc64da385328c6c35eaf1e138eb8734ae99af616bcdd-index.json
40c40
<     "linux-x64-node-v18-{\"/@esbuild/linux-arm64/0.17.6\":{},\"/@esbuild/linux-x64/0.17.6\":{}}": {
---
>     "linux-arm64-node-v18-{\"/@esbuild/linux-arm64/0.17.6\":{},\"/@esbuild/linux-x64/0.17.6\":{}}": {
52c52
<         "integrity": "sha512-20Aodjh/lyF1R1hZL2+UgmNTJFYS2QjdX2xzq+tfpAoUXFy3xAP9eXjU6k9TnYwo3/UUrGf1/3iJZQv0UgmjCQ==",
---
>         "integrity": "sha512-RKT4I4swW4DYYbzIMS7anpJ9G4YErmoIi498d4bFY8E4xTeyQp6mtzwpzS0OtKXeY7Kn2TcJTvLiFXXkLqvOTw==",
54c54
<         "size": 8994816
---
>         "size": 8323072
diff '--color=auto' -r /nix/store/ggnzrp25am692b887qcidb5bb7qbc6x2-en-croissant-pnpm-deps-0.8.0/v3/files/71/eab1a1e754adc6b287b63b657e8d75b6c3cc644e8b2541802e0fae225384129254f5a79c51d90247c8d65253f101643b01f8a8e4b6489912e30be91a5f01a4-index.json /nix/store/s8xvcm8n4dk2qi6p4hrpcr07k0sx1wkr-en-croissant-pnpm-deps-0.8.0/v3/files/71/eab1a1e754adc6b287b63b657e8d75b6c3cc644e8b2541802e0fae225384129254f5a79c51d90247c8d65253f101643b01f8a8e4b6489912e30be91a5f01a4-index.json
40c40
<     "linux-x64-node-v18-{\"/@esbuild/linux-arm64/0.18.20\":{},\"/@esbuild/linux-x64/0.18.20\":{}}": {
---
>     "linux-arm64-node-v18-{\"/@esbuild/linux-arm64/0.18.20\":{},\"/@esbuild/linux-x64/0.18.20\":{}}": {
52c52
<         "integrity": "sha512-VhbLe+ls+gVl6qq+U5OjKNAJP/0019tO9M0eFdaBiwSrwAq0Hv3jNjq3MYAqDGo1yzl59eh9CDRzShH80qPDRw==",
---
>         "integrity": "sha512-1YG98Tqp0Je/JnIVh8CNiw7IYikWAd0iQAiAuV1MrSZs5ZIrvSGElgEkDwH70yp3VKWlWQp1pRRX1/93qYYgzg==",
54c54
<         "size": 9347072
---
>         "size": 8650752
diff '--color=auto' -r /nix/store/ggnzrp25am692b887qcidb5bb7qbc6x2-en-croissant-pnpm-deps-0.8.0/v3/files/92/eb4f7911969bc579765b453f5cc68d038403368b663e12b5d4215956677c6142bc03d5bc787caa4e5b46b5382979d14513678b9b4a408f4f040748e8c3fc9d-index.json /nix/store/s8xvcm8n4dk2qi6p4hrpcr07k0sx1wkr-en-croissant-pnpm-deps-0.8.0/v3/files/92/eb4f7911969bc579765b453f5cc68d038403368b663e12b5d4215956677c6142bc03d5bc787caa4e5b46b5382979d14513678b9b4a408f4f040748e8c3fc9d-index.json
19540c19540
<     "linux-x64-node-v18-{\"/@fortawesome/fontawesome-common-types/6.4.0\":{}}": {
---
>     "linux-arm64-node-v18-{\"/@fortawesome/fontawesome-common-types/6.4.0\":{}}": {
diff '--color=auto' -r /nix/store/ggnzrp25am692b887qcidb5bb7qbc6x2-en-croissant-pnpm-deps-0.8.0/v3/files/97/b89f7d0a53d8ead97c7dab5c8a7bfc592669e64cd2fc5bc4df4a7e20b0198b99ab3320fc976f5f6e91df56719ff3f1becae0dd8f4de974dc9e49ea0f2cafe3-index.json /nix/store/s8xvcm8n4dk2qi6p4hrpcr07k0sx1wkr-en-croissant-pnpm-deps-0.8.0/v3/files/97/b89f7d0a53d8ead97c7dab5c8a7bfc592669e64cd2fc5bc4df4a7e20b0198b99ab3320fc976f5f6e91df56719ff3f1becae0dd8f4de974dc9e49ea0f2cafe3-index.json
40c40
<     "linux-x64-node-v18-{\"/@esbuild/linux-arm64/0.19.8\":{},\"/@esbuild/linux-x64/0.19.8\":{}}": {
---
>     "linux-arm64-node-v18-{\"/@esbuild/linux-arm64/0.19.8\":{},\"/@esbuild/linux-x64/0.19.8\":{}}": {
52c52
<         "integrity": "sha512-j6UwdVufgcjslI7IRBq/yBXBqTlx4BPf6uhcceCAh2yq0W2nV04LwRyHMybczyLLuSm219JYPnzriJW7i5nvxg==",
---
>         "integrity": "sha512-beuCuXDhkQAo6N4WSp5BV2Aod7h19MQMeLZYjtbYL6rpWBhDosaO6kykoQZNZohM2DuavRvUBopRiH1KHnzZtg==",
54c54
<         "size": 9486336
---
>         "size": 8781824
diff '--color=auto' -r /nix/store/ggnzrp25am692b887qcidb5bb7qbc6x2-en-croissant-pnpm-deps-0.8.0/v3/files/ed/d2a04f2c498e5accc056586f56ae8f75ead03f1905efb9a1e2207cc951534e5da17672a5825ed0b024d328e3dd675bc18ce917e4f53cbc704d68b3234a5c93-index.json /nix/store/s8xvcm8n4dk2qi6p4hrpcr07k0sx1wkr-en-croissant-pnpm-deps-0.8.0/v3/files/ed/d2a04f2c498e5accc056586f56ae8f75ead03f1905efb9a1e2207cc951534e5da17672a5825ed0b024d328e3dd675bc18ce917e4f53cbc704d68b3234a5c93-index.json
90c90
<     "linux-x64-node-v18-{\"/@swc/core-linux-arm64-gnu/1.3.100\":{},\"/@swc/core-linux-arm64-musl/1.3.100\":{},\"/@swc/core-linux-x64-gnu/1.3.100\":{},\"/@swc/core-linux-x64-musl/1.3.100\":{},\"/@swc/counter/0.1.2\":{},\"/@swc/types/0.1.5\":{}}": {
---
>     "linux-arm64-node-v18-{\"/@swc/core-linux-arm64-gnu/1.3.100\":{},\"/@swc/core-linux-arm64-musl/1.3.100\":{},\"/@swc/core-linux-x64-gnu/1.3.100\":{},\"/@swc/core-linux-x64-musl/1.3.100\":{},\"/@swc/counter/0.1.2\":{},\"/@swc/types/0.1.5\":{}}": {
```

**Comment by Misterio77** [comment]
It seems that is caused by esbuild. Maybe a good enough solution is per-platform hashes: like this: https://github.com/NixOS/nixpkgs/blob/8ecf09944fe1cf1ce5b08d34dbcd134e5764f0f1/pkgs/by-name/ki/kiwitalk/package.nix#L67

**Comment by lcscosta** [comment]
> It seems that it's caused by esbuild. Maybe a good enough solution is per-platform hashes, like this:
> 
> https://github.com/NixOS/nixpkgs/blob/8ecf09944fe1cf1ce5b08d34dbcd134e5764f0f1/pkgs/by-name/ki/kiwitalk/package.nix#L67-L73


Yes, I have the same doubts about esbuild. Another way to deal with this is by building esbuild from source and using it, but I failed to implement this solution. One example is:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/pot/default.nix#L101-L112

@Misterio77 I attempted to implement a per-platform solution using the hash that worked on my machine, but it failed on ofborg.

My machine metadata
```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.78, NixOS, 23.11 (Tapir), 23.11.20240217.84d981b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by kirillrdy** [comment]
```
$ nix build --nix-path nixpkgs=/home/kirillvr/.cache/nixpkgs-review/pr-271493-1/nixpkgs nixpkgs-overlays=/tmp/tmpxdr1gzfs --extra-experimental-features nix-command no-url-literals --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed -f /home/kirillvr/.cache/nixpkgs-review/pr-271493-1/build.nix
error: hash mismatch in fixed-output derivation '/nix/store/nn0lnxdkia0b6dlp4rvnyilygkfqv249-en-croissant-pnpm-deps-0.9.2.drv':
         specified: sha256-Q1VyEUFeUrcyGkMDl360839QHfzSbbWthqj6akYaP+k=
            got:    sha256-FFVkz/oSXJPB03lCkysGTSkrpdvEly15Y7JTrGKPp/I=
error: 1 dependencies of derivation '/nix/store/n8941dkgx0ds5cai9d86vyjnlivg2c4a-en-croissant-0.9.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/3cfvb5qn8dfcz0fmy6qhn2aaw40y4hnf-review-shell.drv' failed to build

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/271493

1 package failed to build:
en-croissant

error: build log of '/nix/store/n8941dkgx0ds5cai9d86vyjnlivg2c4a-en-croissant-0.9.2.drv^*' is not available
error: build log of '/nix/store/njn6hg4g2y3r346aca7dby5j6xfi5r2n-en-croissant-0.9.2' is not available
```

@lcscosta are you able to diff old FOD and current ?

File: pkgs/by-name/en/en-croissant/package.nix [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+, webkitgtk
+}:
+
+stdenv.mkDerivation rec {
+  pname = "en-croissant";
+  version = "0.9.2";
+
+  src = fetchFromGitHub {
+    owner = "franciscoBSalgueiro";
+    repo = pname;
```
Comment (eclairevoyant) [inline pkgs/by-name/en/en-croissant/package.nix:27]:
```suggestion
    repo = "en-croissant";
```

File: pkgs/by-name/en/en-croissant/package.nix [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+        cpu = [ "x64" "arm64" ];
+      };
+    };
+
+    postPatch = ''
+      mv package.json package.json.orig
+      jq --raw-output ". * $pnpmPatch" package.json.orig > package.json
+    '';
+
+    installPhase = ''
```
Comment (eclairevoyant) [inline pkgs/by-name/en/en-croissant/package.nix:57]: please keep the pre-/post-phase hooks
https://nix.dev/tutorials/packaging-existing-software#phases-and-hooks

File: pkgs/by-name/en/en-croissant/package.nix [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+
+  preInstall = ''
+    mv target/release/bundle/deb/*/data/usr/ $out
+  '';
+
+  meta = with lib; {
+    description = "A Modern Chess Database";
+    homepage = "https://github.com/franciscoBSalgueiro/en-croissant/";
+    maintainers = [ maintainers.lcscosta ];
+    license = licenses.gpl3;
```
Comment (eclairevoyant) [inline pkgs/by-name/en/en-croissant/package.nix:119]: this is a deprecated identifier; please use gpl3Plus or gpl3Only in accordance with whichever upstream went with

File: pkgs/by-name/en/en-croissant/package.nix [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
```
Comment (eclairevoyant) [inline pkgs/by-name/en/en-croissant/package.nix:120]: as it's now built from source, should we remove this now?

File: pkgs/by-name/en/en-croissant/package.nix [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+        sed -i -E -e 's/"checkedAt":[0-9]+,//g' $f
+        jq --sort-keys . $f | sponge $f
+      done
+    '';
+
+    dontFixup = true;
+    outputHashMode = "recursive";
+    outputHash = {
+      x86_64-linux = "sha256-Q1VyEUFeUrcyGkMDl360839QHfzSbbWthqj6akYaP+k=";
+    }.${stdenv.system} or (throw "Unsupported system: ${stdenv.system}");
```
Comment (eclairevoyant) [inline pkgs/by-name/en/en-croissant/package.nix:75]: does this mean only `x86_64-linux` is supported for this package?

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
Thank you for the contribution and your work so far! I've provided some feedback below:

**Comment by SuperSandro2000** [comment]
#290715 got now merged

**Comment by gepbird** [comment]
Superseded by #341451.

**Pull request closed without merging**